File: mobile/lib/features/banners/domain/entities/banner.dart
import 'package:equatable/equatable.dart';

class BannerEntity extends Equatable {
  final String id;
  final String salesId;
  final String? customerId;
  final String? leadId;
  final String shopName;
  final String content;
  final String dimensions;
  final String? photoPath;
  final double latitude;
  final double longitude;
  final String? address;
  final DateTime createdAt;
  final DateTime updatedAt;

  const BannerEntity({
    required this.id,
    required this.salesId,
    this.customerId,
    this.leadId,
    required this.shopName,
    required this.content,
    required this.dimensions,
    this.photoPath,
    required this.latitude,
    required this.longitude,
    this.address,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        salesId,
        customerId,
        leadId,
        shopName,
        content,
        dimensions,
        photoPath,
        latitude,
        longitude,
        address,
        createdAt,
        updatedAt,
      ];
}

File: backend/internal/domain/models/banner.go
package models

import (
	"crm_wowin_backend/pkg/utils"
	"github.com/google/uuid"
)

type Banner struct {
	ID         uuid.UUID     `json:"id"`
	SalesID    uuid.UUID     `json:"sales_id"`
	CustomerID *uuid.UUID    `json:"customer_id,omitempty"`
	LeadID     *uuid.UUID    `json:"lead_id,omitempty"`
	ShopName   string        `json:"shop_name"`
	Content    string        `json:"content"`
	Dimensions string        `json:"dimensions"`
	PhotoPath  *string       `json:"photo_path,omitempty"`
	Latitude   float64       `json:"latitude"`
	Longitude  float64       `json:"longitude"`
	Address    *string       `json:"address,omitempty"`
	CreatedAt  utils.FlexTime `json:"created_at"`
	UpdatedAt  utils.FlexTime `json:"updated_at"`
}
